plugins/backend/mock/mock-plugin/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"os"
+
+	"github.com/hashicorp/vault/helper/pluginutil"
+	"github.com/hashicorp/vault/logical/plugin"
+	"github.com/hashicorp/vault/plugins/backend/mock"
+)
+
+func main() {
+	apiClientMeta := &pluginutil.APIClientMeta{}
+	flags := apiClientMeta.FlagSet()
+	flags.Parse(os.Args)
+	tlsConfig := apiClientMeta.GetTLSConfig()
+	tlsProviderFunc := pluginutil.VaultPluginTLSProvider(tlsConfig)
+
+	plugin.Serve(&plugin.ServeOpts{
```

Comment:
I think backend plugins and database plugins should enter at the same `Serve` function. Right now database plugins call `/plugins/serve.go Serve()` could backend plugins have the same entry point?